Server/src/models/DataUploadModel.ts

```diff
@@ -15,190 +15,185 @@ import { Material } from './entities/Material';
-        await connection.manager.save(someMaterial);
+            let someMaterial = new Material();
+            someMaterial.id;
+            someMaterial.compositionId = composition.id;
+            someMaterial.details = material[i].details;
+            await connection.manager.save(someMaterial);
+        }
     }
-}
 
-const insertCategories = async (_category, sub_category) => {
+    async insertCategories(_category: string, _subCategory: string) {
 
-    const connection = getConnection();
-    let categoryIDs = [];
+        const connection = getConnection();
+        let categoryIDs = [];
 
-    let category = new Category();
-    category.id;
-    category.name = _category;
-    await connection.manager.save(category);
+        let category = new Category();
+        category.id;
+        category.name = _category;
+        await connection.manager.save(category);
 
-    let subcategory = new Subcategory();
-    subcategory.id;
-    subcategory.name = sub_category;
-    await connection.manager.save(subcategory);
-    categoryIDs.push(category.id, subcategory.id);
-    return categoryIDs;
-}
+        let subcategory = new Subcategory();
+        subcategory.id;
+        subcategory.name = _subCategory;
+        await connection.manager.save(subcategory);
+        categoryIDs.push(category.id, subcategory.id);
+        return categoryIDs;
+    }
 
-const insertDataSetDataType = async (datasetdatatypeValue) => {
+    async insertDataSetDataType(datasetdatatypeValue: string) {
 
-    const connection = getConnection();
+        const connection = getConnection();
 
-    let datasetdatatype = new Datasetdatatype();
-    datasetdatatype.id;
-    datasetdatatype.name = datasetdatatypeValue;
-    await connection.manager.save(datasetdatatype);
-    console.log("datasetdatatypeValue saved.");
-    return datasetdatatype.id;
-}
+        let datasetdatatype = new Datasetdatatype();
+        datasetdatatype.id;
+        datasetdatatype.name = datasetdatatypeValue;
+        await connection.manager.save(datasetdatatype);
+        console.log("datasetdatatypeValue saved.");
+        return datasetdatatype.id;
+    }
 
-const insertFullDataSet = async (dataSetName, dataSetDataTypeID, publicationID, /* categoryIDs ,*/ material, dataSetComments) => {
-
-    const connection = getConnection();
-
-    let dataset = new Dataset();
-    dataset.id;
-    dataset.name = dataSetName;
-    dataset.datatypeId = dataSetDataTypeID;
-    dataset.publicationId = publicationID;
-    dataset.categoryId; //= categoryIDs[0];
-    dataset.subcategoryId; //= categoryIDs[1];
-    dataset.materials = material;
-    dataset.comments = dataSetComments;
-    await connection.manager.save(dataset);
-    console.log("First dataset saved.");
-    return dataset.id;
-}
+    async insertFullDataSet(dataSetName: string, dataSetDataTypeID: number, publicationID: number, /* categoryIDs ,*/ material: any[], dataSetComments: string) {
```

Comment:
add a descriptive todo for  /* categoryIDs ,*/ or remove it

Reply:
Perhaps briefly mention the category/subcategory insertion issues at the current time?